xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/common/LedgerSpecifier.java

```diff
@@ -0,0 +1,197 @@
+   *
+   * @param ledgerHashHandler          A {@link Consumer} that is called if {@link #ledgerHash()} is present.
+   * @param ledgerIndexHandler         A {@link Consumer} that is called if {@link #ledgerIndex()} is present.
+   * @param ledgerIndexShortcutHandler A {@link Consumer} that is called if {@link #ledgerIndexShortcut()}
+   *                                   is present.
+   */
+  @Value.Auxiliary
+  default void handle(
+    final Consumer<Hash256> ledgerHashHandler,
+    final Consumer<LedgerIndex> ledgerIndexHandler,
+    final Consumer<LedgerIndexShortcut> ledgerIndexShortcutHandler
+  ) {
+    Objects.requireNonNull(ledgerHashHandler);
+    Objects.requireNonNull(ledgerIndexHandler);
+    Objects.requireNonNull(ledgerIndexShortcutHandler);
+
+    ledgerHash().ifPresent(ledgerHashHandler);
+    ledgerIndex().ifPresent(ledgerIndexHandler);
+    ledgerIndexShortcut().ifPresent(ledgerIndexShortcutHandler);
+  }
+
+  /**
+   * Map this {@link LedgerSpecifier} to an instance of {@link R}, depending on which specifier is present.
+   *
+   * @param ledgerHashMapper          A {@link Function} that is called if {@link #ledgerHash()} is present.
+   * @param ledgerIndexMapper         A {@link Function} that is called if {@link #ledgerIndex()} is present.
+   * @param ledgerIndexShortcutMapper A {@link Function} that is called if {@link #ledgerIndexShortcut()}
+   *                                  is present.
+   * @param <R>                       The type of object to return after mapping.
+   *
+   * @return A {@link R} that is constructed by the appropriate mapper function.
+   */
+  @Value.Auxiliary
+  default <R> R map(
+    final Function<Hash256, R> ledgerHashMapper,
+    final Function<LedgerIndex, R> ledgerIndexMapper,
+    final Function<LedgerIndexShortcut, R> ledgerIndexShortcutMapper
+  ) {
+    Objects.requireNonNull(ledgerHashMapper);
+    Objects.requireNonNull(ledgerIndexMapper);
+    Objects.requireNonNull(ledgerIndexShortcutMapper);
+
+    if (ledgerHash().isPresent()) {
+      return ledgerHashMapper.apply(ledgerHash().get());
+    } else if (ledgerIndex().isPresent()) {
+      return ledgerIndexMapper.apply(ledgerIndex().get());
+    } else if (ledgerIndexShortcut().isPresent()) {
+      return ledgerIndexShortcutMapper.apply(ledgerIndexShortcut().get());
+    } else {
+      throw new IllegalStateException("Unsupported field.");
+    }
+  }
+
+  /**
+   * Validates that only one of the three fields in a {@link LedgerSpecifier} is present.
+   */
+  @Value.Check
+  default void validateOnlyOneSpecified() {
+    int numSpecified = 0;
+    if (ledgerHash().isPresent()) {
+      numSpecified += 1;
+    }
+    if (ledgerIndex().isPresent()) {
+      numSpecified += 1;
+    }
+    if (ledgerIndexShortcut().isPresent()) {
+      numSpecified += 1;
+    }
+
+    Preconditions.checkArgument(
+      numSpecified == 1,
+      String.format("Exactly one Ledger specifier must be provided. %s were specified", numSpecified)
```

Comment:
This error message isn't going to make sense to a developer without them going and inspecting our source code to see what it means.

Could we make it a bit more descriptive, maybe like this:

```suggestion
      String.format("Only one Ledger specifier may be specified using one of ledgerHash, ledgerIndex or ledgerIndexShortcut. %s were specified", numSpecified)
```